java/src/main/java/com/google/api/services/samples/youtube/cmdline/live/ListLiveChatMessages.java

```diff
@@ -0,0 +1,207 @@
+     */
+    private static void listChatMessages(
+        final String liveChatId,
+        final String nextPageToken,
+        long delayMs) {
+        System.out.println(
+            String.format("Getting chat messages in %1$.3f seconds...", delayMs * 0.001));
+        Timer pollTimer = new Timer();
+        pollTimer.schedule(
+            new TimerTask() {
+                @Override
+                public void run() {
+                    try {
+                        // Get chat messages from YouTube
+                        LiveChatMessageListResponse response = youtube
+                            .liveChatMessages()
+                            .list(liveChatId, "snippet, authorDetails")
+                            .setPageToken(nextPageToken)
+                            .setFields(LIVE_CHAT_FIELDS)
+                            .execute();
+
+                        // Display messages and super chat details
+                        List<LiveChatMessage> messages = response.getItems();
+                        for (int i = 0; i < messages.size(); i++) {
+                            LiveChatMessage message = messages.get(i);
+                            LiveChatMessageSnippet snippet = message.getSnippet();
+                            System.out.println(buildOutput(
+                                snippet.getDisplayMessage(),
+                                message.getAuthorDetails(),
+                                snippet.getSuperChatDetails()));
+                        }
+
+                        // Request the next page of messages
+                        listChatMessages(
+                            liveChatId,
+                            response.getNextPageToken(),
+                            response.getPollingIntervalMillis());
+                    } catch (Throwable t) {
+                        System.err.println("Throwable: " + t.getMessage());
+                        t.printStackTrace();
+                    }
+                }
+            }, delayMs);
+    }
+
+    /**
+     * Formats a chat message for console output.
+     *
+     * @param message The display message to output.
+     * @param author The author of the message.
+     * @param superChatDetails SuperChat details associated with the message.
+     * @return A formatted string for console output.
+     */
+    private static String buildOutput(
+        String message,
+        LiveChatMessageAuthorDetails author,
+        LiveChatSuperChatDetails superChatDetails) {
+        StringBuilder output = new StringBuilder();
+        if (superChatDetails != null) {
+            output.append(superChatDetails.getAmountDisplayString());
+            output.append("SUPERCHAT RECEIVED FROM ");
+        }
+        output.append(author.getDisplayName());
+        if (author.getIsChatOwner() || author.getIsChatModerator() || author.getIsChatSponsor()) {
+            output.append(" (");
+            boolean appendComma = false;
```

Comment:
Two things:
1. If Java supports a string.join method, it might be easier to just add "OWNER|MODERATOR|SPONSOR" to an array. Pseudo-code for lines 179-200:
roles = new Array[];
if author.getIsChatOwner() {
  roles.append("Owner");
}
if author.getIsChatModerator() {
  roles.append("Moderator");
}
if author.getIsChatSponsor() {
  roles.append("Sponsor");
}
if roles.length > 0 {
  output.append(" (" + roles.join(", ") + ")");
}

2. If every owner is a moderator, I'd make line 186 an "else if" clause, in which case the display possibilities will be:
(OWNER, SPONSOR)
(MODERATOR, SPONSOR)
(OWNER)
(MODERATOR)
(SPONSOR)

instead of:
(OWNER, MODERATOR, SPONSOR)
(MODERATOR, SPONSOR)
(OWNER)
(MODERATOR)
(SPONSOR)

But, if you can be an owner and not a moderator, then keep the code as is.